native/common.hpp

```diff
@@ -133,6 +134,69 @@ dm::NumericTablePtr make_table(T *data, size_t rows, size_t cols) {
 }
 
 
+/*
+ * Write a NumericTable to npy format.
+ */
+template <typename T>
+void write_table(dm::NumericTablePtr table,
+                 const std::string descr, const std::string fn) {
+
+    dm::BlockDescriptor<T> block;
+    table->getBlockOfRows(0, table->getNumberOfRows(), dm::readOnly, block);
+    T *data = block.getBlockPtr();
+
+    size_t shape[] = {table->getNumberOfRows(), table->getNumberOfColumns()};
+
+    char *c_descr = new char[descr.size() + 1];
+    strcpy(c_descr, descr.c_str());
+    const struct npyarr arr {
+        .descr = c_descr,
+        .fortran_order = false,
+        .shape_len = 2,
+        .shape = shape,
+        .data = data
+    };
+
+    save_npy(&arr, fn.c_str(), sizeof(T));
+
+    table->releaseBlockOfRows(block);
+
+}
+
+
+/*
+ * Create a new HomogenNumericTable from a submatrix of an existing
+ * HomogenNumericTable.
+ */
+template <typename T>
+dm::NumericTablePtr
+copy_submatrix(dm::NumericTablePtr src,
+               size_t row, size_t col, size_t rows, size_t cols) {
```

Comment:
Names should be more descriptive: `size_t row` -> `size_t row_start`, `size_t rows` -> `size_t num_rows`, etc.

Reply:
Or better yet, the input should be `row_start`, `row_finish`, consistent with NumPy slices, then `num_rows = row_finish - row_start`.

Reply:
Fixed. It's now consistent with numpy slices.